coder_sniffer/Drupal/Sniffs/Classes/ClassFileNameSniff.php

```diff
@@ -57,6 +57,20 @@ public function process(File $phpcsFile, $stackPtr)
             return ($phpcsFile->numTokens + 1);
         }
 
+        // If the file is not a known php file (like a txt file or a md file),
+        // we do not need to check if the file is named after the class we
+        // found.
+        $extension     = pathinfo($fullPath, PATHINFO_EXTENSION);
+        $phpExtensions = [
+            'inc',
+            'php',
+            'module',
+            'theme',
```

Comment:
I think we should only check *.php files. Those are the only files where we want to compare file names with class names.

Then we can also remove the special handling for behat.inc files below. It was never the intention to test special behat files with this sniff anyway. This is for PSR4 *.php files and their names.